core/src/migrations/000097-addStateToImports.ts

```diff
@@ -0,0 +1,71 @@
+import Sequelize from "sequelize";
+
+export default {
+  up: async (
+    queryInterface: Sequelize.QueryInterface,
+    DataTypes: typeof Sequelize
+  ) => {
+    await queryInterface.renameColumn(
+      "imports",
+      "recordUpdatedAt",
+      "importedAt"
+    );
+
+    await queryInterface.removeColumn("imports", "groupsUpdatedAt");
+
+    await queryInterface.addColumn("imports", "state", {
+      type: DataTypes.STRING(191),
+      allowNull: true,
+    });
+
+    await queryInterface.sequelize.query(
+      `UPDATE "imports" SET "state"='associating' WHERE "recordAssociatedAt" IS NULL`
+    );
+
+    await queryInterface.sequelize.query(
+      `UPDATE "imports" SET "state"='importing' WHERE "recordAssociatedAt" IS NOT NULL AND "importedAt" IS NULL`
+    );
+
+    await queryInterface.sequelize.query(
+      `UPDATE "imports" SET "state"='exporting' WHERE "recordAssociatedAt" IS NOT NULL AND "importedAt" IS NOT NULL AND "exportedAt" IS NULL`
+    );
+
+    await queryInterface.sequelize.query(
+      `UPDATE "imports" SET "state"='complete' WHERE "recordAssociatedAt" IS NOT NULL AND "importedAt" IS NOT NULL AND "exportedAt" IS NOT NULL`
+    );
+
+    await queryInterface.sequelize.query(
+      `UPDATE "imports" SET "state"='failed' WHERE "errorMessage" IS NOT NULL`
+    );
+
+    await queryInterface.changeColumn("imports", "state", {
+      type: DataTypes.STRING(191),
+      allowNull: false,
+    });
```

Comment:
👍 we end up with a non-null state column

Reply:
Do you think we need any addtional exclusivity checks on the updates, eg:

```sql
UPDATE "imports" SET "state"='exporting' WHERE "recordAssociatedAt" IS NOT NULL AND "importedAt" IS NOT NULL AND "exportedAt" IS NULL ((AND state IS NULL))
```

Reply:
👍 this will fail to migrate if there are any imports that still don't have a state

Reply:
Added `state IS NULL` checks to the migration on ec560ca